instagram_scraper.py

```diff
@@ -17,88 +18,132 @@
+                        total_count = 0
+                        break
+                continue
+            imgs.add(url)
+            hashtags = set(REGEXES['hashtag'].findall(caption))
+            mentions = set(REGEXES['username'].findall(caption))
+            count += 1
+            yield url, caption, hashtags, mentions
+
+
+
+def scrape_instagram(target: str, total_count: int=50, existing: set=None, mode: str='tags'):
     """
-    :param tags:
-        List of tags that need to be scraped.
+    :param targets:
+        List of targets that need to be scraped.
     :param total_count:
         Total number of images to be scraped.
+    :param existing:
+        URLs to skip
+    :param mode
+        Two options: 'tags' or 'users'. Determines whether we are scraping users or tags
+        
+    Builds url and sets short_circuit based on target and then issues request to url
     """
-    for tag in tags:
-        yield from scrape_instagram_tag(tag, total_count)
+    if mode == 'users':
+        short_circuit = True
+        url = f'https://www.instagram.com/{target}'
+    else:
+        short_circuit = False
+        url = f'https://www.instagram.com/explore/tags/{target}'
+
+    yield from send_scrape_request(url, total_count=total_count, existing=existing, short_circuit=short_circuit)
 
-def main(tags, total_count, should_continue):
-    def _single_tag_processing(tag, total_count, existing_links, start):
-        os.makedirs(f'data/{tag}', exist_ok=True)
-        with open(f'data/{tag}/data.csv', 'a' if existing_links else 'w', newline='', encoding='utf-8') as csvfile:
+
+def main(tags: List[str], users: List[str], total_count: int=50, should_continue: bool=False):
+    """
+    :param tags:
+        List of tags to be scraped
+    :param users:
+        List of users to be scraped
+    :param total_count:
+        total number of images to be scraped
+    :param should_continue
+        Flag for whether or not we should read from disk and skip existing URLs
+        
+    Scrapes user and hashtag images from Instagram
+    """
+    def _single_input_processing(target: str, total_count: int, existing_links: set, start: int, mode: str='tag'):
+        os.makedirs(f'data/{target}', exist_ok=True)
+        with open(f'data/{target}/data.csv', 'a' if existing_links else 'w', newline='', encoding='utf-8') as csvfile:
             writer = csv.writer(csvfile, delimiter=',')
-            for count, (url, caption, hashtags, mentions) in enumerate(scrape_instagram_tag(
-                tag, total_count, existing_links), start):
+            for count, (url, caption, hashtags, mentions) in enumerate(scrape_instagram(
+                target, total_count, existing_links, mode), start):
 
                 try:
                     req = requests.get(url)
-                    with open(f'data/{tag}/{count}.jpg', 'wb') as img:
+                    with open(f'data/{target}/{count}.jpg', 'wb') as img:
```

Comment:
We want the users to be able to distinguish between the `user` photos, and `tag` photos, since if I scrape `@instagram`, I might mistake it for images scraped from `instagram` tag. So, mode specific data directories. :)